### DISCUSSION TITLE
LIS SURFACEMODEL outputs not giving GPP and LAI output

### DISCUSSION THREAD

**Discussion by Arijit22IITDelhi** [NASA-LIS/LISF#1493] [Q&A] [answered]
Hello, I was trying to run LIS to produce GPP and LAI outputs along with other land surface state variables. 
This is the corresponding **LDT config file** using which I generated the LIS input file:
[LDT_Punjab.txt](https://github.com/NASA-LIS/LISF/files/14340498/LDT_Punjab.txt)
In the **output attributes table**, I also enabled both of the parameters:
[OUTPUT_Attribs.txt](https://github.com/NASA-LIS/LISF/files/14340519/OUTPUT_Attribs.txt)
**LIS config file:** 
[Config.txt](https://github.com/NASA-LIS/LISF/files/14340547/Config.txt)

But after running Noah MP 3.6 as LSM within LIS with the above configuration, I analyzed the outputs but it is showing all 0 values for GPP and LAI.
![GPP](https://github.com/NASA-LIS/LISF/assets/137404009/a97b46d6-4d3a-4a24-b4ef-8029b75b9252)
![LAI](https://github.com/NASA-LIS/LISF/assets/137404009/76949e60-e16b-4553-b1cb-16153287c6e7)

Please suggest me what necessary changes I have to make in any of these files to get GPP and LAI outputs.

**Comment by dmocko** [comment, marked as answer]
HI @Arijit22IITDelhi 

In Noah-MP, you can only get GPP/NPP/NEE, or LeafMass/RootMass/StemMass/WoodMass/DeepSoilCarbon/ShallowSoilCarbon outputs when you turn on the dynamic vegetation scheme.  In Noah-MP-3.6, this is done with this setting:

`Noah-MP.3.6 vegetation model option:             2         # dveg_opt: vegetation model`

Your LAI output is zero because you have set the the LAI data source as:

`LAI data source:                        none`

Note that if you do use this dynamic vegetation option in Noah-MP-3.6, the LAI and Greenness will be simulated by the dynamic vegetation physics in the model.

**Reply by Arijit22IITDelhi** [reply]
Thank you so much. So if I select dynamic vegetation option, do I need to specify LAI option too (as LDT) ? Or It will still give LAI output even if I keep LAI data source as "none"?

**Reply by dmocko** [reply]
Good question.  Sorry that I didn't clarify.  If you keep `vegetation model option` as "1", you will need to specify the `LAI data source` as "LDT" to get LAI output.  If you change `vegetation model option` to "2", I would have `LAI data source` as "none", so that the dynamic vegetation scheme will instead prognostically calculate/update the LAI.

**Reply by Arijit22IITDelhi** [reply]
@dmocko Thanks a lot.

**Reply by Arijit22IITDelhi** [reply]
@dmocko I was also trying to get Potential Evapotranspiration output but whenever I tried to run after turning on the PET option in output attribute table it fails. How can I get Potential Evapotranspiration output apart from total evapotranspiration?

**Reply by dmocko** [reply]
Noah-MP does not calculate PET, so it's not available as an output.  That's a feature of the model formulation, and unrelated to an issue of the LSM's LIS implementation.  You can see what variables are available to be output from each LSM by looking at the "_main.F90" code within lis/surfacemodels/land/[LSM_name].

**Reply by Arijit22IITDelhi** [reply]
Ok, got it. Thank you so much again.
